JULIERAJ/KIDS-FIRST-Back
# feat/8/Add checkAuth API endpoint to protect routes

PR Template Backend

One Line Description
  - Add checkAuth API endpoint for route protection.

Description
  - Implemented a checkAuth api to verify user authentication status and protect routes. The endpoint returns a 401 Unauthorized status and message if no user is authenticated. This aligns with the frontend task to ensure unauthorized users are redirected to the login page when attempting to access protected routes.


Notes
  - This PR enhances security by aligning backend route protection with frontend expectations.
  - Ensure testing covers both backend authentication checks and frontend redirection behaviors.

Test Steps
- Testing Instructions for Postman

1. Start the backend server.

2. First try to access a protected route before being authenticated:

      - Endpoint: /api/logout
              Method: GET
              Headers: None
              Expected Response: 401 Unauthorized with message "No token provided"

3. Then login to get authentication

      - Endpoint: /api/login
              Method: POST
              Headers: Content-Type: application/json
              Body:
              {
                "email": "user@example.com",
                "password": "password123"
              }
              Expected Response: 200 OK with user data and authentication token

4. Then try the same protected route after being authenticated

      - Endpoint: /api/logout
              Method: GET
              Headers: None
              Expected Response: 200 OK with protected data

API Changes (if there are any)
Added checkAuth to verify user authentication status.


Checklist
